File: mobileBE/src/main/java/com/uth/mobileBE/dto/response/UserResponse.java
package com.uth.mobileBE.dto.response;


import com.uth.mobileBE.models.enums.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserResponse {
    private Long userId;
    private String username;
    private String fullname;
    private Role role;
    private Boolean isActive;
    private LocalDateTime createdAt;
    private LocalDateTime updateAt;
}

File: app/src/main/java/com/example/quanlythuvien/data/model/response/UserResponse.kt
package com.example.quanlythuvien.data.model.response

import com.google.gson.annotations.SerializedName

data class UserResponse(
    @SerializedName("userId")
    val userId: Long,

    @SerializedName("username")
    val username: String,

    @SerializedName("fullname")
    val fullname: String,

    @SerializedName("role")
    val role: String, // ADMIN, STAFF

    @SerializedName("isActive")
    val isActive: Boolean,

    @SerializedName("createdAt")
    val createdAt: String,

    @SerializedName("updateAt")
    val updateAt: String?
)